lib/dl_zitadel/dl_zitadel/middlewares/aiohttp.py

```diff
@@ -21,6 +23,123 @@
 class AioHTTPMiddleware:
     _client: clients.ZitadelAsyncClient
     _token_storage: services.ZitadelAsyncTokenStorage
+    _allow_user_auth: bool = attr.ib(default=False)
+    _allow_service_auth: bool = attr.ib(default=True)
+
+    async def auth_user(
+        self,
+        app_request: dl_api_commons_aiohttp_aiohttp_wrappers.DLRequestBase,
+    ) -> middlewares_models.AuthResult | None:
+        req = app_request.request
+        zitadel_cookie_str = req.cookies.get(middlewares_models.ZitadelCookies.ZITADEL_COOKIE)
+        if zitadel_cookie_str is None:
+            LOGGER.info("Zitadel cookie is missing")
+            return None
+        zitadel_cookie = json.loads(base64.b64decode(zitadel_cookie_str))
+        self._zitadel_cookie_storage = zitadel_cookie
+
+        try:
+            access_token, refresh_token = (
+                zitadel_cookie["passport"]["user"]["accessToken"],
+                zitadel_cookie["passport"]["user"]["refreshToken"],
+            )
+        except KeyError:
+            access_token, refresh_token = None, None
+
+        if access_token is None:
+            LOGGER.info("Access token is missing")
+            return None
+
+        token_introspect_result = await self._client.introspect(token=access_token)
+
+        if not token_introspect_result.active:
+            LOGGER.info("Access token is not active")
+
+            if refresh_token is None:
+                LOGGER.info("Refresh token is missing")
+                return None
+
+            refreshed_tokens = await self._client.refresh_token(refresh_token)
+            LOGGER.info("Tokens successfully refreshed")
+            access_token, refresh_token = (
+                refreshed_tokens.access_token,
+                refreshed_tokens.refresh_token,
+            )
+
+            token_introspect_result = await self._client.introspect(token=access_token)
+
+        else:
+            LOGGER.info("Access token is active")
+
+        zitadel_cookie["passport"]["user"]["accessToken"] = access_token
+        zitadel_cookie["passport"]["user"]["refreshToken"] = refresh_token
+
+        zitadel_cookie = base64.b64encode(json.dumps(zitadel_cookie).encode("ascii")).decode("ascii")
+
+        return middlewares_models.AuthResult(
+            user_id=token_introspect_result.sub,
+            user_name=token_introspect_result.username,
+            data=middlewares_models.ZitadelAuthData(
+                service_access_token=await self._token_storage.get_token(),
+                user_access_token=access_token,
+            ),
+            updated_cookies=zitadel_cookie,
```

Comment:
updated_cookies - better use extendable dict[str, str] and pass
`{middlewares_models.ZitadelCookies.ZITADEL_COOKIE: zitadel_cookie}`